lib/internal/util/comparisons.js

```diff
@@ -345,6 +349,21 @@ function getEnumerables(val, keys) {
   return ArrayPrototypeFilter(keys, (key) => hasEnumerable(val, key));
 }
 
+function partialSymbolEquiv(val1, val2, keys2) {
+  const symbolKeys = getOwnSymbols(val2);
+  if (symbolKeys.length !== 0) {
+    for (const key of symbolKeys) {
+      if (hasEnumerable(val2, key)) {
+        if (!hasEnumerable(val1, key)) {
+          return false;
+        }
+        ArrayPrototypePush(keys2, key);
```

Comment:
is push faster in this case, or would `keys2[keys2.length] = key;` be faster?

Reply:
I have to check. This specific case is pretty rare, though. We normally do not have symbols to compare, so I'll keep it for now.